app/tests/test_app.py

```diff
@@ -1,96 +1,74 @@
-from fastapi.testclient import TestClient
-from sqlalchemy import create_engine
-from sqlalchemy.orm import sessionmaker
+from http import HTTPStatus
 
-from app.database import Base
-from app.main import app, get_session
-
-SQLALCHEMY_DATABASE_URL = "sqlite:///./test.db"
-
-engine = create_engine(
-    SQLALCHEMY_DATABASE_URL, connect_args={"check_same_thread": False}
-)
-TestingSessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
 
+def test_should_create_a_book(client):
+    # given
+    payload = {"title": "test_title1", "author": "test_author1", "price": 9.99}
+    # when
+    response = client.post("/books/", json=payload)
+    # then
+    data = response.json()
+    assert response.status_code == HTTPStatus.OK.value, response.text
+    payload.update({"id": 1})
+    expected_data = payload
+    assert data == expected_data
 
-Base.metadata.create_all(bind=engine)
 
+def test_should_get_all_books(client):
+    # when
+    response = client.get("/books/")
```

Comment:
what if we decided to change our url?

Reply:
added reverse url lookups in tests to avoid this problem